compiler/src/dotty/tools/dotc/transform/DispatchToSpecializedApply.scala

```diff
@@ -0,0 +1,29 @@
+package dotty.tools
+package dotc
+package transform
+
+import TreeTransforms.{ MiniPhaseTransform, TransformerInfo }
+import core._
+import Contexts.Context, Types._, Decorators._, Symbols._, DenotTransformers._
+import Denotations._, SymDenotations._, Scopes._, StdNames._, NameOps._, Names._
+
+class DispatchToSpecializedApply extends MiniPhaseTransform {
+  import ast.Trees._
+  import ast.tpd
+
+  val phaseName = "dispatchToSpecializedApply"
+
+  override def transformApply(tree: tpd.Apply)(implicit ctx: Context, info: TransformerInfo) =
+    tree match {
+      case Apply(select @ Select(id, nme.apply), arg :: Nil) =>
+        val params = List(arg.tpe, tree.tpe)
+        val specializedApply = nme.apply.specializedFor(params, params.map(_.typeSymbol.name))
```

Comment:
yes) would be nice to have an extra boolean arg that checks if it should force creation.